### PR TITLE
Implement a time update() function for the gravity function plugin

### PR DESCRIPTION
**Pull request by jaustermann**
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

*Describe what you did in this PR and why you did it.*

I added a time update() function to the gravity function. I did that following the implementation in other function plugins. I did not test this addition because it's not immediately obvious to me how it would best be tested but am happy to do so if you have suggestions on how this would be done. This addresses issue #3341. 

### For all pull requests:

* [X] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/master/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/master/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

**Comment by tjhei** [comment]
Your email address in git is not set correctly.

**Comment by jaustermann** [comment]
Did that fix it? Git is behaving kind of weird and doesn't update my name and email even though the right ones show up in my config. But, it did revert to my old email address, which is what I think I've used before. Let me know.

File: include/aspect/gravity_model/function.h [hunk 1]
```diff
@@ -39,14 +40,22 @@ namespace aspect
...
       public:
         /**
          * Constructor.
          */
         Function ();
 
+        /**
+         * A function that is called at the beginning of each time step to
+         * indicate what the model time is for which the boundary values will
+         * next be evaluated. For the current class, the function passes to
```
Comment (gassmoeller) [inline include/aspect/gravity_model/function.h:54]:
```suggestion
         * indicate what the model time is for which the gravity will
         * next be evaluated. For the current class, the function passes to
```

**Comment by jaustermann** [comment]
Thanks Rene, am doing this now. Do you want me to have separate commits for the test and changelog entry or squash it all together?

**Pull request merged**